src/container/__tests__/RelayContainer_setVariables-test.js

```diff
@@ -792,5 +792,99 @@ describe('RelayContainer.setVariables', function() {
         size: 48,
       });
     });
+
+    it('does not reset variables if outside props are the same', () => {
+      class MockInnerComponent extends React.Component {
+        render() {
+          return <div />;
+        }
+      }
+
+      const MockInnerContainer = Relay.createContainer(MockInnerComponent, {
+        fragments: {
+          entity: () => Relay.QL`  fragment on User {
+                        url(site:$site)
+                        storySearch(query:$query) {
+                          id
+                        }
+                        profilePicture(size:$size) {
+                          uri
+                        }
+                      }`,
+        },
+        initialVariables: {
+          site: 'mobile',
+          query: undefined, // <-- Object type
+          size: undefined // <-- Array type
+        },
+      });
+
+      class MockWrapperComponent extends React.Component {
+        render() {
+          return (
+            <MockInnerContainer
+              ref='inner'
+              query={this.props.relay.variables.query}
+              size={this.props.relay.variables.size}
+              entity={this.props.entity}
+            />
+          );
+        }
+      }
+
+      MockContainer = Relay.createContainer(MockWrapperComponent, {
```

Comment:
Hey I made this change on the new test I created by creating a new variable and not using the `MockContainer` variable declared on the top level describe. I assume it was done this way for speed maybe? Let me know your thoughts.

Also, there are still some other test on this file that store the container of `MockWrapperComponent` in the top level `MockContainer` variable, should I go ahead and update those tests as well?